api/CMakeLists.txt

```diff
@@ -35,7 +35,7 @@ if(WITH_NO_DEPRECATED_CODE)
                              INTERFACE OPENTELEMETRY_NO_DEPRECATED_CODE)
 endif()
 
-if(WITH_ABSEIL)
+if(WITH_ABSEIL OR WITH_OTLP_GRPC)
```

Comment:
@owent It might be better not to enforce HAVE_ABSEIL at this point. Perhaps we can address it in a separate PR, allowing this one to focus solely on sharing the gRPC client. What do you think?

Reply:
I got a crash before, in which case some codes are compiled without `HAVE_ABSEIL`, but the codes in gRPC exporters must be compiled with it. I think it's the simplest way to keep compatibility to enforce abseil-cpp when using gRPC exporters now.
I think we can remove it if we change the include paths of all bundled version of abseil-cpp, or there will be conflicts.( Some codes in gRPC exporters will be compiled with the bundled version of abseil-cpp because of the search order of headers, and the gRPC library is built with the external version)

Reply:
Is the crash related to code within this PR. As there is no crash observed before that ?

Reply:
This PR will use the new headers which also includes abseil-cpp's headers in some versions of gRPC.Without this macro, gRPC's headers may include the bundled version of abseil-cpp in otel-cpp.
The codes before did not use these new headers but there are still problems when otel-cpp's api use headers from the bundled version, but gRPC exporters use the external version. I think it's UB, and may lead to export nothing when using gRPC exporter.

In my understanding, the best solution is to change the file paths of bundled version of abseil, so the two versions will not share the same include path.

Reply:
Does this only apply to protobuf 3.21 and below? With protobuf 3.22, the SDK build will fail if `WITH_OTLP_GRPC` is enabled without `WITH_ABSEIL`.

Reply:
This is not related to protobuf, gRpc always depends abseil-cpp. gRPC depends external abseil-cpp.
Maybe we can let all gRPC exporters always search headers from third party's include paths first so they will not use headers from bundled version.

Reply:
https://github.com/open-telemetry/opentelemetry-cpp/actions/runs/11596474550/job/32287629537?pr=3041
The conflicts are show here, can I remove inline namespace for bundled absl to avoid conflicts?